subiquity/client/client.py

```diff
@@ -227,6 +227,22 @@ async def noninteractive_confirmation(self):
             answer = await run_in_thread(input)
         await self.confirm_install()
 
+    async def _status_get(self, cur=None):
+        while True:
+            try:
+                return await self.client.meta.status.GET(cur=cur)
+            except aiohttp.ClientError:
+                try:
+                    fp = open(self.state_path("server-state"))
+                except FileNotFoundError:
+                    pass
+                else:
+                    with fp:
+                        state = getattr(ApplicationState, fp.read(), None)
+                    if state == ApplicationState.EXITED:
+                        self.exit()
```

Comment:
do we need handling for the non-exited case here?
presumably we're only in this part if the server was running, and the process has exited without cleaning transitioning to an EXITED state first - a server process crash most likely.
FWIW, I tried to create this scenario by running dryrun and stepping thru a few screens then killing the server process, and created a different client crash instead
```
  File "/home/dbungert/dev/github/canonical/subiquity/subiquity/common/api/client.py", line 82, in make_request
    async with session.request(
  File "/usr/lib/python3/dist-packages/aiohttp/client.py", line 1117, in __aenter__
    self._resp = await self._coro
  File "/usr/lib/python3/dist-packages/aiohttp/client.py", line 520, in _request
    conn = await self._connector.connect(
  File "/usr/lib/python3/dist-packages/aiohttp/connector.py", line 535, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/usr/lib/python3/dist-packages/aiohttp/connector.py", line 1197, in _create_connection
    raise ClientConnectorError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorError: Cannot connect to host a:80 ssl:default [Connection refused]
```

Reply:
We also have to consider that the server might be restarting. We could be more explicit about that by adding a RESTARTING state and explicitly not-handling that here I suppose.

It's a bit poor that killing the server makes the client crash but I'm not sure it's that bad either. That's a different codepath of course and so best handled in a different branch?

Reply:
Yes, new branch is fine

Reply:
Or the client crash thing on a possible future branch I mean